### ISSUE TITLE
Error computing kernel of group hom from fp group into perm group

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#1774]
The following issue was reported by Graham Ellis on the support list:
```
gap> a:=(2,6,7,5,4,3,8)(9,32,21,10,25,19,14,31,17,11,29,22,13,26,23,15,27,24,12,28,20)(16,30,18);;
gap> G:=Group([a]);;
gap> F:=FreeGroup(3);;
gap> f:=GroupHomomorphismByImages(F,G,GeneratorsOfGroup(F),[a,a,a]);;
gap> Kernel(f);
Error, List Element: <list>[4] must have an assigned value in
 return Image( e1, Ag[i] ) * Image( e2, Bg[i] )
; at /usr/local/lib/gap4r8/lib/grpfp.gi:1936 called from 
func( C[i] ) at /usr/local/lib/gap4r8/lib/coll.gi:746 called from
List( [ 1 .. Length( Ag ) ], function ( i )
     return Image( e1, Ag[i] ) * Image( e2, Bg[i] );
 end ) at /usr/local/lib/gap4r8/lib/grpfp.gi:1936 called from
Intersection2( I, D ) at /usr/local/lib/gap4r8/lib/coll.gi:2490 called from
Intersection( cor, u ) at /usr/local/lib/gap4r8/lib/ghomfp.gi:136 called from
KernelOfMultiplicativeGeneralMapping( map 
) at /usr/local/lib/gap4r8/lib/overload.g:311 called from
....  at line 5 of *stdin*
you can 'return;' after assigning a value
brk> 
```
But this works:
```
gap> a:=(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21);;
gap> G:=Group([a]);;
gap> F:=FreeGroup(3);;
gap> f:=GroupHomomorphismByImages(F,G,GeneratorsOfGroup(F),[a,a,a]);;
gap> Kernel(f);
Group(<free, no generators known>)
```

The issue is *not* present in GAP 4.4.12, but it *is* present in GAP 4.6 -- I did not try to bisect it further.

**Comment by fingolfin** [comment]
Looking closer at the error, it occurs in the Intersection2 method for "subgroups of fp group by quotient", which contains this code:
```
...
  A:=G!.quot;
  B:=H!.quot;
...
  Ag:=GeneratorsOfGroup(A);
  Bg:=GeneratorsOfGroup(B);
  # form the sdp
  sg:=List([1..Length(Ag)],i->Image(e1,Ag[i])*Image(e2,Bg[i]));  # <-- error is here
...
```
Now, the two groups A and B are in representation `IsSubgroupOfWholeGroupByQuotientRep`, and as I understand it, this means that the groups `A` (i.e. `G!.quot`) and `B` (i.e. `H!.quot`) should have the same number of generators as the parent group `F` of `G` and `H`, namely 3. But `A` actually has 6 generators, which leads to the error.

Perhaps @hulpke has some further insight on this?

**Comment by alex-konovalov** [comment]
If the fix is imminent, i suggest to make GAP 4.8.9 - there are a few packages waiting for the next minor release already.

**Comment by hulpke** [comment]
#1809 includes a fix. (The problem was that `Image' of a group does not guarantee that the same generators are used.)
